pandas/core/internals/blocks.py

```diff
@@ -1669,7 +1669,11 @@ def _unwrap_setitem_indexer(self, indexer):
             #  Should never have length > 2.  Caller is responsible for checking.
             #  Length 1 is reached vis setitem_single_block and setitem_single_column
             #  each of which pass indexer=(pi,)
-            if len(indexer) == 2:
+            if len(indexer) == 1:
```

Comment:
Not sure, but it might make more sense to handle this in __setitem__ of the ArrowExtensionArray?

Reply:
I was 50/50 on whether to put it here or in ArrowExtensionArray.__setitem__. I ended up leaning towards this as it seems consistent with this method and may be helpful beyond ArrowExtensionArray (e.g. third-party EAs). But happy to move it if setitem is a better place for it.

Reply:
I am more concerned with our setiteim for Arrow diverging from the numpy setitem call, but not sure if this is correct either.

cc @jbrockmendel Could you weigh in here?

Reply:
id want to handle it correctly in the ArrowArray method.  could add a test to the extension base tests for handling this case correctly

Reply:
ok, I will change this to be handled within `ArrowExtensionArray.__setitem__`.  

I guess the typing on `ExtensionArray.__setitem__` and `ArrowExtensionArray.__setitem__` should change as well?

https://github.com/pandas-dev/pandas/blob/14881577a2bd874ee45dde7230cf334b69f9026e/pandas/core/arrays/base.py#L351

https://github.com/pandas-dev/pandas/blob/14881577a2bd874ee45dde7230cf334b69f9026e/pandas/core/arrays/arrow/array.py#L878

Reply:
Yeah would make sense